src/org/labkey/filetransfer/view/FileTransferMetadataQueryView.java

```diff
@@ -18,40 +18,58 @@
 import org.jetbrains.annotations.Nullable;
 import org.labkey.api.data.ActionButton;
 import org.labkey.api.data.ButtonBar;
+import org.labkey.api.data.Container;
 import org.labkey.api.data.DataRegionSelection;
 import org.labkey.api.data.TableInfo;
 import org.labkey.api.exp.list.ListDefinition;
 import org.labkey.api.query.QuerySettings;
-import org.labkey.api.query.QueryView;
 import org.labkey.api.query.UserSchema;
+import org.labkey.api.study.Study;
+import org.labkey.api.study.StudyService;
+import org.labkey.api.util.Button;
+import org.labkey.api.util.PageFlowUtil;
 import org.labkey.api.util.StringExpression;
 import org.labkey.api.util.StringExpressionFactory;
 import org.labkey.api.view.ActionURL;
 import org.labkey.api.view.DataView;
 import org.labkey.api.view.Portal;
+import org.labkey.api.view.SimpleTextDisplayElement;
+import org.labkey.api.view.ViewContext;
 import org.labkey.filetransfer.FileTransferController;
 import org.labkey.filetransfer.FileTransferManager;
 import org.labkey.filetransfer.provider.FileTransferProvider;
 import org.labkey.filetransfer.provider.Registry;
 import org.labkey.filetransfer.query.FileTransferMetadataTable;
 import org.labkey.filetransfer.query.FileTransferQuerySchema;
+import org.labkey.study.model.DatasetDefinition;
+import org.labkey.study.query.DatasetQuerySettings;
+import org.labkey.study.query.DatasetQueryView;
+import org.springframework.validation.BindException;
 import org.springframework.validation.Errors;
+import org.apache.log4j.Logger;
+import org.springframework.validation.ObjectError;
+import org.springframework.web.servlet.ModelAndView;
+import org.springframework.web.servlet.View;
 
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
 import java.util.Map;
 
 import static org.labkey.filetransfer.FileTransferManager.FILE_TRANSFER_PROVIDER;
 
 /**
  * Created by susanh on 5/9/17.
  */
-public class FileTransferMetadataQueryView extends QueryView
+// public class FileTransferMetadataQueryView extends QueryView
+public class FileTransferMetadataQueryView extends DatasetQueryView
```

Comment:
This is Dennis' area, so I can't speak to how or why it works, but it definitely builds fine; I think it's because we actually build this module as part of a much larger trialshare project.  All of this has been running smoothly on our demo server for a while now.  Do you foresee any potential problems?

Reply:
Yes, this is definitely a problem.  It does not build for me, and I generally would  not expect it to build.  You are likely not encountering a problem because you already have the study jar in the compile path for a different reason.

Reply:
There is a dependency on Study.  In build.gradle:

import org.labkey.gradle.util.BuildUtils

dependencies
        {
            external 'com.google.oauth-client:google-oauth-client:1.22.0'
            BuildUtils.addLabKeyDependency(project: project, config: "compile", depProjectPath: ":server:api")
            BuildUtils.addLabKeyDependency(project: project, config: "compile", depProjectPath: ":server:modules:study")
        }

Reply:
In order to keep interfaces between modules clear, we generally do not do this.  

Also, the :server:api dependency should already have been provided by the application of the JavaPlugin.

Reply:
Good to know.  We'll be looking at doing some major changes in the Spring to bring the dataset out of a webpart and into a more dataset-like view, so we should revisit this then and find the best way to do it.